### PR TITLE
feat: Unify dialogs across app - part 2

### PR DESCRIPTION
**Pull request by lrupnikk**

### CODE REVIEW

File: brd-android/fabriik-common/src/main/java/com/fabriik/common/ui/dialog/FabriikGenericDialog.kt [hunk 1]
```diff
@@ -63,11 +63,21 @@ class FabriikGenericDialog : DialogFragment() {
     }
 
     private fun setupDescription(view: TextView) {
-        val description = if (args.descriptionRes != null) getString(args.descriptionRes!!) else args.description
+        val description = if (args.descriptionRes != null) getStringRes() else args.description
         view.text = description
         view.isVisible = description != null
     }
 
+    private fun getStringRes(): String {
```
Comment (roknovak) [inline brd-android/fabriik-common/src/main/java/com/fabriik/common/ui/dialog/FabriikGenericDialog.kt:71]: This is a bad solution, we need to allow undefined number of arguments. You can do something similar:

```suggestion
    private fun getStringRes(): String {
        val messageArgs = args.messageArgs?.toTypedArray() ?: emptyArray()
        return getString(args.descriptionRes!!, *messageArgs)
    }
```

File: brd-android/app/src/main/java/com/breadwallet/ui/importwallet/ImportController.kt [hunk 2]
```diff
@@ -108,6 +111,26 @@ class ImportController(
...
+                CONFIRM_IMPORT_DIALOG_POSITIVE ->
+                    eventConsumer.accept(E.OnImportConfirm)
+                CONFIRM_IMPORT_DIALOG_NEGATIVE ->
+                    eventConsumer.accept(E.OnImportCancel)
+            }
+        }
+
+        registerForGenericDialogResult(IMPORT_SUCCESS_DIALOG) { resultKey, _ ->
+            when (resultKey) {
+                IMPORT_SUCCESS_DIALOG_POSITIVE ->
```
Comment (roknovak) [inline brd-android/app/src/main/java/com/breadwallet/ui/importwallet/ImportController.kt:129]: Check this callback please, I think you forgot to handle negative button result

File: brd-android/app/src/main/java/com/breadwallet/ui/settings/SettingsController.kt [hunk 3]
```diff
@@ -129,6 +130,13 @@ class SettingsController(
                     eventConsumer.accept(E.OnTestGenericDialogResult("Test dialog positive button clicked"))
             }
         }
+
+        registerForGenericDialogResult(CONFIRM_EXPORT_TRANSACTIONS_DIALOG) {resultKey, _ ->
+            when (resultKey) {
+                TEST_FABRIIK_DIALOG_POSITIVE ->
```
Comment (roknovak) [inline brd-android/app/src/main/java/com/breadwallet/ui/settings/SettingsController.kt]: Wrong result key

**Review by roknovak — APPROVED** [review]

**Review by dabitdev — APPROVED** [review]
LGTM

**Pull request merged**